### PR TITLE
Fix minor inconsistencies in event types

### PR DESCRIPTION
**Pull request by Saalvage**
## Description
`SDL_CameraDeviceEvent` was missing a reserved field and 2 specific events as well as the event union itself were using `Uint32` instead of `SDL_EventType`.

### CODE REVIEW

**Comment by sezero** [comment]
To fix the `-Wswitch` errors, you should do the following:
```diff
diff --git a/test/testime.c b/test/testime.c
index aa25e9d..57e4f9f 100644
--- a/test/testime.c
+++ b/test/testime.c
@@ -775,6 +775,9 @@ int main(int argc, char *argv[])
                 cursor = event.edit.start;
                 Redraw();
                 break;
+
+            default:
+                break;
             }
         }
     }
```

File: include/SDL3/SDL_events.h [hunk 1]
```diff
@@ -680,7 +681,7 @@ typedef struct SDL_QuitEvent
  */
 typedef struct SDL_UserEvent
 {
-    Uint32 type;        /**< ::SDL_EVENT_USER through ::SDL_EVENT_LAST-1 */
+    SDL_EventType type; /**< ::SDL_EVENT_USER through ::SDL_EVENT_LAST-1 */
```
Comment (sezero) [inline include/SDL3/SDL_events.h:684]: Not sure about the others, but I guess this will be a problem in c++, no?

Comment (Saalvage) [reply include/SDL3/SDL_events.h:684]: I'm not exactly sure if this is the problem you're considering, but it seems like it shouldn't be an issue: https://stackoverflow.com/questions/33812998/is-it-allowed-for-an-enum-to-have-an-unlisted-value

**Comment by slouken** [comment]
I think I've pushed commits covering all the feedback in this PR, can this be closed?

**Comment by Saalvage** [comment]
You've added the notes to `SDL_Event` and `SDL_UserEvent`, while the PR comment refers to `SDL_CommonEvent`. I believe `SDL_CommonEvent` should feature the same note?

**Comment by slouken** [comment]
> You've added the notes to `SDL_Event` and `SDL_UserEvent`, while the PR comment refers to `SDL_CommonEvent`. I believe `SDL_CommonEvent` should feature the same note?

Yes, thanks for catching that!

**Pull request closed without merging**